runtime/common/src/parachains.rs

```diff
@@ -640,6 +640,7 @@ impl<T: Trait> Module<T> {
 
 		for head in heads.iter() {
 			let id = head.parachain_index();
+			Heads::insert(id, &head.candidate.head_data.0);
```

Comment:
A somewhat random place to insert this. Should probably be in the `check_candidates` loop instead.

Reply:
I just added it back to the position where it was before being removed :woman_shrugging: 

https://github.com/paritytech/polkadot/commit/d9b4fc4572bd6a24501b96e2ba5293bc87fd2f72#diff-e21f8096708b22fcdbd66ebc74f0821aL457

Reply:
It looks like `Heads` should be modified only in case `check_candidates` succeeded; if that insert was moved in `check_candidates` loop over all candidates we may end up with a partially modified `Heads` ? Or add another loop in the end of `check_candidates` right before returning `Ok`.